lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -72,20 +69,7 @@ class FullPageScreenshot extends Gatherer {
    * @return {Promise<LH.Artifacts.FullPageScreenshot | null>}
    */
   async afterPass_(passContext) {
-    let screenshot = await this._takeScreenshot(passContext, MAX_SCREENSHOT_HEIGHT);
-
-    if (screenshot.data.length > MAX_DATA_URL_SIZE) {
-      // Hitting the data URL size limit is rare, it only happens for pages on tall
-      // desktop sites with lots of images.
-      // So just cutting down the height a bit usually fixes the issue.
-      screenshot = await this._takeScreenshot(passContext, 5000);
-      if (screenshot.data.length > MAX_DATA_URL_SIZE) {
-        passContext.LighthouseRunWarnings.push(
-          'Full page screenshot is too big–report won\'t show element screenshots.');
-        return null;
-      }
-    }
-
+    let screenshot = await this._takeScreenshot(passContext);
```

Comment:
nit: Can we remove the `afterPass_` function and call `_takeScreenshot` directly?